File: Process_Back/back_03_SystemManagement/back_SystemSetting03_Environment/back_EnvironmentSetting.py
from publicUseFunction import importPackage, dbConnection
back_EnvironmentSetting = importPackage.Blueprint(
    'back_EnvironmentSetting', __name__)
conn = importPackage.conn

pool = importPackage.psycopg2.pool.ThreadedConnectionPool(
    minconn=1, maxconn=999999999999, dsn=importPackage.DB_URI)


@back_EnvironmentSetting.route('/backEnvironmentSetting', methods=['GET', 'POST'])
def backEnvironmentSetting():
    user_id = importPackage.session.get('user_id')
    user = importPackage.session.get('user_name')
    current_time = importPackage.datetime.datetime.now()
    formatted_time = current_time.strftime("%Y/%m/%d")

    conn = pool.getconn()

    user_id = importPackage.session.get('user_id')
    user_name = importPackage.session.get('user_name')
    user_email = importPackage.session.get('user_email')

    exam_Registration = importPackage.session.get('exam_Registration')
    agency_Registration = importPackage.session.get('agency_Registration')

    system_accountAdd = importPackage.session.get('system_accountAdd')
    system_accountModify = importPackage.session.get('system_accountModify')
    system_passwordReset = importPackage.session.get('system_passwordReset')
    system_Environment = importPackage.session.get('system_Environment')
    system_SMTPSetting = importPackage.session.get('system_SMTPSetting')
    system_SystemSearch = importPackage.session.get('system_SystemSearch')

    basic_CarouselSetting = importPackage.session.get('basic_CarouselSetting')
    basic_FastLinkSetting = importPackage.session.get('basic_FastLinkSetting')
    basic_NewSetting = importPackage.session.get('basic_NewSetting')
    basic_NormalQuestionSetting = importPackage.session.get(
        'basic_NormalQuestionSetting')
    basic_NoticeSetting = importPackage.session.get('basic_NoticeSetting')
    basic_PaymentSetting = importPackage.session.get('basic_PaymentSetting')
    basic_UnitImageSetting = importPackage.session.get(
        'basic_UnitImageSetting')

    try:
        with conn.cursor() as cursor:
            cursor.execute(""" select * from "back_EnvironmentSetting" """)
            result = cursor.fetchall()
    except Exception as e:
        result = None
        print('Error')
    finally:
        pool.putconn(conn)

    if importPackage.request.method == "POST":

        switchcarouselsetting = importPackage.request.form['hiddenswitchcarouselsetting']
        UpLoadTime = importPackage.request.form['hiddenUpLoadTime']
        RemoveTime = importPackage.request.form['hiddenRemoveTime']
        inputWebTitle = importPackage.request.form['hiddeninputWebTitle']
        inputEmailTitle = importPackage.request.form['hiddeninputEmailTitle']
        inputDescription = importPackage.request.form['hiddeninputDescription']
        inputAddress = importPackage.request.form['hiddeninputAddress']
        inputswitchboard = importPackage.request.form['hiddeninputswitchboard']
        inputFax = importPackage.request.form['hiddeninputFax']
        inputPrivacy = importPackage.request.form['hiddeninputPrivacy']
        inputuseraccount = importPackage.request.form['hiddenuseraccount']
        conn = pool.getconn()
        try:
            with conn.cursor() as cursor:
                cursor.execute("""
                    update "back_EnvironmentSetting"
                            SET "switchcarouselsetting"=%s,"UpLoadTime"=%s,"RemoveTime"=%s,
                            "maintainTitle"=%s,"maintainEmail"=%s,"maintainContext"=%s,"inputAddress"=%s,
                            "inputswitchboard"=%s,"inputFax"=%s,"inputPrivacy"=%s
                    where "inputuseraccount"=%s """,
                               (switchcarouselsetting, UpLoadTime, RemoveTime,
                                inputWebTitle, inputEmailTitle, inputDescription, inputAddress,
                                inputswitchboard, inputFax, inputPrivacy, inputuseraccount))
                conn.commit()
        except Exception as e:
            print('Error:', e)
        finally:
            pool.putconn(conn)

    return importPackage.render_template('後台網頁/back_03_System_Management/back_03_Environment/Back_EnvironmentSetting.html', result=result, userid=user_id, Handler=user_name, DateTime=formatted_time,
                                         user_id=user_id,
                                         user_name=user_name,
                                         user_email=user_email,
                                         exam_Registration=exam_Registration,
                                         agency_Registration=agency_Registration,
                                         system_accountAdd=system_accountAdd,
                                         system_accountModify=system_accountModify,
                                         system_passwordReset=system_passwordReset,
                                         system_Environment=system_Environment,
                                         system_SMTPSetting=system_SMTPSetting,
                                         system_SystemSearch=system_SystemSearch,
                                         basic_CarouselSetting=basic_CarouselSetting,
                                         basic_FastLinkSetting=basic_FastLinkSetting,
                                         basic_NewSetting=basic_NewSetting,
                                         basic_NormalQuestionSetting=basic_NormalQuestionSetting,
                                         basic_NoticeSetting=basic_NoticeSetting,
                                         basic_PaymentSetting=basic_PaymentSetting,
                                         basic_UnitImageSetting=basic_UnitImageSetting)

File: static/後台網頁/back_03_System_Management/back_03_Environment/back_EnvironmentSetting.js
document.addEventListener("DOMContentLoaded", function() {
    var switchcarouselsetting = document.querySelector('input[name="switchcarouselsetting"]');
    var hiddenswitchcarouselsetting = document.querySelector('input[name="hiddenswitchcarouselsetting"]');
    // 將 checkbox 的值設置到 hiddenInput 的值中
    hiddenswitchcarouselsetting.value = switchcarouselsetting.checked ? "true" : "false";
    // 監聽 checkbox 的改變事件
    switchcarouselsetting.addEventListener('change', function() {
    // 將 checkbox 的值設置到 hiddenInput 的值中
    hiddenswitchcarouselsetting.value = switchcarouselsetting.checked ? "true" : "false";
    });
});


document.addEventListener("DOMContentLoaded", function() {
    var UpLoadTime = document.querySelector('input[name="UpLoadTime"]');
    var hiddenUpLoadTime = document.querySelector('input[name="hiddenUpLoadTime"]');
    hiddenUpLoadTime.value = UpLoadTime.value;
    UpLoadTime.addEventListener('input', function() {
        hiddenUpLoadTime.value = UpLoadTime.value;
    });
});

document.addEventListener("DOMContentLoaded", function() {
    var RemoveTime = document.querySelector('input[name="RemoveTime"]');
    var hiddenRemoveTime = document.querySelector('input[name="hiddenRemoveTime"]');
    hiddenRemoveTime.value = RemoveTime.value;
    RemoveTime.addEventListener('input', function() {
        hiddenRemoveTime.value = RemoveTime.value;
    });
});

document.addEventListener("DOMContentLoaded", function() {
    var inputWebTitle = document.querySelector('input[name="inputWebTitle"]');
    var hiddeninputWebTitle = document.querySelector('input[name="hiddeninputWebTitle"]');
    hiddeninputWebTitle.value = inputWebTitle.value;
    inputWebTitle.addEventListener('input', function() {
        hiddeninputWebTitle.value = inputWebTitle.value;
    });
});

document.addEventListener("DOMContentLoaded", function() {
    var inputEmailTitle = document.querySelector('input[name="inputEmailTitle"]');
    var hiddeninputEmailTitle = document.querySelector('input[name="hiddeninputEmailTitle"]');
    hiddeninputEmailTitle.value = inputEmailTitle.value;
    inputEmailTitle.addEventListener('input', function() {
        hiddeninputEmailTitle.value = inputEmailTitle.value;
    });
});

document.addEventListener("DOMContentLoaded", function() {
    var inputDescription = document.querySelector('input[name="inputDescription"]');
    var hiddeninputDescription = document.querySelector('input[name="hiddeninputDescription"]');
    hiddeninputDescription.value = inputDescription.value;
    inputDescription.addEventListener('input', function() {
        hiddeninputDescription.value = inputDescription.value;
    });
});

document.addEventListener("DOMContentLoaded", function() {
    var inputAddress = document.querySelector('input[name="inputAddress"]');
    var hiddeninputAddress = document.querySelector('input[name="hiddeninputAddress"]');
    hiddeninputAddress.value = inputAddress.value;
    inputAddress.addEventListener('input', function() {
        hiddeninputAddress.value = inputAddress.value;
    });
});

document.addEventListener("DOMContentLoaded", function() {
    var inputswitchboard = document.querySelector('input[name="inputswitchboard"]');
    var hiddeninputswitchboard = document.querySelector('input[name="hiddeninputswitchboard"]');
    hiddeninputswitchboard.value = inputswitchboard.value;
    inputswitchboard.addEventListener('input', function() {
        hiddeninputswitchboard.value = inputswitchboard.value;
    });
});

document.addEventListener("DOMContentLoaded", function() {
    var inputFax = document.querySelector('input[name="inputFax"]');
    var hiddeninputFax = document.querySelector('input[name="hiddeninputFax"]');
    hiddeninputFax.value = inputFax.value;
    inputFax.addEventListener('input', function() {
        hiddeninputFax.value = inputFax.value;
    });
});

document.addEventListener("DOMContentLoaded", function() {
    var inputPrivacy = document.querySelector('input[name="inputPrivacy"]');
    var hiddeninputPrivacy = document.querySelector('input[name="hiddeninputPrivacy"]');
    hiddeninputPrivacy.value = inputPrivacy.value;
    inputPrivacy.addEventListener('input', function() {
        hiddeninputPrivacy.value = inputPrivacy.value;
    });
});

document.addEventListener("DOMContentLoaded", function() {
    var normalQuestionTitle= document.querySelector('input[name="normalQuestionTitle"]');
    var hiddennormalQuestionTitle= document.querySelector('input[name="hiddennormalQuestionTitle"]');
    
    // 初始化时将normalQuestionTitle的值填入hiddennormalQuestionTitle
    hiddennormalQuestionTitle.value = normalQuestionTitle.value;
    
    // 监听normalQuestionTitle的输入事件，保持两者的值同步
    normalQuestionTitle.addEventListener('input', function() {
        hiddennormalQuestionTitle.value = normalQuestionTitle.value;
    });
});